subsys/net/lib/prometheus/formatter.c

```diff
@@ -0,0 +1,237 @@
+/*
+ * Copyright (c) 2024 Mustafa Abdullah Kus, Sparse Technology
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <zephyr/net/prometheus/formatter.h>
+#include <zephyr/net/prometheus/collector.h>
+#include <zephyr/net/prometheus/metric.h>
+#include <zephyr/net/prometheus/histogram.h>
+#include <zephyr/net/prometheus/summary.h>
+#include <zephyr/net/prometheus/gauge.h>
+#include <zephyr/net/prometheus/counter.h>
+
+#include <stdarg.h>
+#include <stdio.h>
+#include <string.h>
+#include <stddef.h>
+#include <errno.h>
+
+#include <zephyr/kernel.h>
+
+#include <zephyr/logging/log.h>
+LOG_MODULE_REGISTER(pm_formatter, CONFIG_PROMETHEUS_LOG_LEVEL);
+
+static int write_metric_to_buffer(char *buffer, size_t buffer_size, const char *format, ...)
+{
+	/* helper function to write formatted metric to buffer */
+	va_list args;
+	size_t len = strlen(buffer);
+
+	if (len >= buffer_size) {
+		return -ENOMEM;
+	}
+	buffer += len;
+	buffer_size -= len;
+
+	va_start(args, format);
+	len = vsnprintf(buffer, buffer_size, format, args);
+	va_end(args);
+	if (len >= buffer_size) {
+		return -ENOMEM;
+	}
+
+	return 0;
+}
+
+int prometheus_format_exposition(const struct prometheus_collector *collector, char *buffer,
+				 size_t buffer_size)
+{
+	int ret;
+	int written = 0;
```

Comment:
Maybe it is thanks to a previous refactoring... It looks like `written` is not updated and is always 0, so could be removed.